codegenerator/cli/templates/static/codegen/src/globalState/GlobalState.res

```diff
@@ -3,6 +3,33 @@ open Belt
 type chain = ChainMap.Chain.t
 type rollbackState = NoRollback | RollingBack(chain) | RollbackInMemStore(InMemoryStore.t)
 
+module WriteDebouncers = {
+  type t = {chainMetaData: Debouncer.t, pruneStaleData: ChainMap.t<Debouncer.t>}
+  let make = (~config: Config.t): t => {
+    let chainMetaData = {
+      let intervalMillis = Env.DebounceWrites.chainMetadataIntervalMillis
+      let logger = Logging.createChild(
+        ~params={
+          "context": "Debouncer for chain metadata writes",
+          "intervalMillis": intervalMillis,
+        },
+      )
+      Debouncer.make(~intervalMillis, ~logger)
```

Comment:
I think this is convenient here, but isn't it more correct to pass the logger from where we call `schedule`, instead of when we create the Debouncer

Reply:
It can work either way, then we just store the logger here in this state and pass where it get's called. I guess it's just a preference of where we store the logger. I think it's better than constructing a new logger on every call unless we want to pass in different context values, which might be nice as well 🤔

Reply:
Actually after looking back at the code, it is best this way.

startInternal will recurse through scheduled calls, so if that takes the logger as a param then it can only hand down the same logger even if it's a different scheduled call. 

If you actually want a logger to be tied to where it gets scheduled then the "scheduled" state should also have a logger. I think we can do that but it's not needed yet.